api/src/main/java/edu/cornell/mannlib/vitro/webapp/auth/attributes/AttributeValueKey.java

```diff
@@ -9,15 +13,31 @@ public class AttributeValueKey {
     private AccessObjectType aot;
     private String role;
     private String type;
+    private Set<String> namedKeyComponents = new HashSet<>();
+
+    public Set<String> getNamedKeyComponents() {
+        return namedKeyComponents;
+    }
 
     public AttributeValueKey() {
     }
 
-    public AttributeValueKey(AccessOperation ao, AccessObjectType aot, String role, String type) {
+    public AttributeValueKey(AccessOperation ao, AccessObjectType aot, String role, String type,
+            String... namedKeyComponents) {
         this.ao = ao;
         this.aot = aot;
         this.role = role;
         this.type = type;
+        this.namedKeyComponents = new HashSet<>(Arrays.asList(namedKeyComponents));
+    }
+
+    public AttributeValueKey(AccessOperation ao, AccessObjectType aot, String role, String type,
+            Set<String> namedKeyComponents) {
+        this.ao = ao;
+        this.aot = aot;
```

Comment:
I would rename class attributes to the full name. Constructor parameters can stay as abbreviations.

Reply:
Fixed